exchange/events_test.go

```diff
@@ -192,3 +195,296 @@ func Test_isEventAllowed(t *testing.T) {
 		})
 	}
 }
+
+func TestConvertToVastEvent(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    config.Events
+		expected injector.VASTEvents
+	}{
+		{
+			name: "Error event",
+			input: config.Events{
+				DefaultURL: "http://default.url",
+				VASTEvents: []config.VASTEvent{
+					{
+						CreateElement:     config.ErrorVASTElement,
+						URLs:              []string{"http://error.url"},
+						ExcludeDefaultURL: true,
+					},
+				},
+			},
+			expected: injector.VASTEvents{
+				TrackingEvents: make(map[string][]string),
+				Errors:         []string{"http://error.url"},
+			},
+		},
+		{
+			name: "NonLinearTracking event",
+			input: config.Events{
+				DefaultURL: "http://default.url",
+				VASTEvents: []config.VASTEvent{
+					{
+						CreateElement:     config.NonLinearClickTrackingVASTElement,
+						URLs:              []string{"http://tracking.url"},
+						ExcludeDefaultURL: true,
+					},
+				},
+			},
+			expected: injector.VASTEvents{
+				TrackingEvents:         make(map[string][]string),
+				NonLinearClickTracking: []string{"http://tracking.url"},
+			},
+		},
+		{
+			name: "CompanionClickThrough event",
+			input: config.Events{
+				DefaultURL: "http://default.url",
+				VASTEvents: []config.VASTEvent{
+					{
+						CreateElement:     config.CompanionClickThroughVASTElement,
+						URLs:              []string{"http://tracking.url"},
+						ExcludeDefaultURL: false,
+					},
+				},
+			},
+			expected: injector.VASTEvents{
+				TrackingEvents:        make(map[string][]string),
+				CompanionClickThrough: []string{"http://tracking.url", "http://default.url"},
+			},
+		},
+		{
+			name: "Tracking event",
+			input: config.Events{
+				DefaultURL: "http://default.url",
+				VASTEvents: []config.VASTEvent{
+					{
+						CreateElement: config.TrackingVASTElement,
+						Type:          "start",
+						URLs:          []string{"http://tracking.url"},
+					},
+				},
+			},
+			expected: injector.VASTEvents{
+				TrackingEvents: map[string][]string{
+					"start": {"http://tracking.url", "http://default.url"},
+				},
+			},
+		},
+		{
+			name: "Click tracking event",
+			input: config.Events{
+				DefaultURL: "http://default.url",
+				VASTEvents: []config.VASTEvent{
+					{
+						CreateElement: config.ClickTrackingVASTElement,
+						URLs:          []string{"http://clicktracking.url"},
+					},
+				},
+			},
+			expected: injector.VASTEvents{
+				TrackingEvents: make(map[string][]string),
+				VideoClicks:    []string{"http://clicktracking.url", "http://default.url"},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := convertToVastEvent(tt.input)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func TestAppendURLs(t *testing.T) {
```

Comment:
Can you add a test case where `urls` is nil and one where `event.URLs` is nil?

Reply:
Done